src/plugins/intel_gna/src/transformations/gather_sinking_unary.hpp

```diff
@@ -0,0 +1,114 @@
+// Copyright (C) 2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include "openvino/pass/graph_rewrite.hpp"
+#include "transformations_visibility.hpp"
+
+namespace ov {
+namespace pass {
+
+class TRANSFORMATIONS_API GatherSinkingUnaryForward;
+class TRANSFORMATIONS_API GatherSinkingUnaryBackwardSingleConsumer;
+class TRANSFORMATIONS_API GatherSinkingUnaryBackwardMultiConsumers;
+class TRANSFORMATIONS_API GatherSinkingUnaryBackward;
+
+}  // namespace pass
+}  // namespace ov
+
+/**
+ * @brief Moves Gather layer forward from the start to the end of the graph
+ * through the unary operations UnaryElementwiseArithmetic, Clamp, Elu, SoftPlus, LogicalNot, Convert
+ * 
+ *  Gather          Unary
+ *    |      =>       |
+ *  Unary           Gather
+ *    |               |
+ *  Another         Another
+ * 
+ *   Gather                Unary
+ *     |            =>       |
+ *   Unary                Gather
+ *    |  |                 |   |
+ * Any1  Any2             Any1 Any2
+ * 
+ *     Gather                Unary1
+ *       |           =>      |   |
+ *     Unary1              Unary2 Unary3
+ *     |    |                |     |
+ * Unary2  Unary3          Gather Gather
+ * 
+ *     Another1              Another1
+ *        |                  |      |
+ *     Gather             Unary   Gather
+ *     |    |         =>     |         |
+ *    Unary Another2       Gather     Another2
+ *     |                     |
+ *    Another3              Another3
+ * 
+ * All GatherSinking tranformations are designed to work in 2 steps:
+ * - forward push
+ * - backward push
+ * Add flag into Gather layer rt_info that prevents backward sinking if the next layer
+ * after Gather does not support by GatherSinking transformations. That is done to
+ * prevent backward pushing the layer that already pushed forward through the graph.
+ */
+class ov::pass::GatherSinkingUnaryForward : public ov::pass::MatcherPass {
```

Comment:
I am new to this kind of transformation and it is not clear for me.
Are they automatically registered or they are currently used only in tests?

Reply:
There is a large task 79521. This is a small part of it. It is really hard to do code review on a huge PR. We are merging small parts. After all code merged we will fix GNAPlugin::LoadNetwork to load this transformation.